app/domain/authentication/authn_oidc/authenticator.rb

```diff
@@ -156,15 +156,23 @@ def audit_failure(err)
         @audit_log.log(
           ::Audit::Event::Authn::Authenticate.new(
             authenticator_name: authenticator_name,
-            service:            webservice,
-            role:               role,
-            client_ip:          client_ip,
-            success:            false,
-            error_message:      err.message
+            service: webservice,
+            role_id: audit_role_id,
```

Comment:
You make a helper method for creating the `RoleId` object in this class but in-line it in `authenticate.rb`. We should probably pick one pattern and try to stick to it.